app/models/user.rb

```diff
@@ -9,6 +9,8 @@ class User < ActiveRecord::Base
   has_many :following_relationships, class_name: "Relationship", foreign_key: :follower_id
   has_many :leading_relationships, class_name: "Relationship", foreign_key: :leader_id
 
+  before_create :generate_token
```

Comment:
You should be creating token upon request - not when user is created. That is a security threat.